ESP/lib/src/data/config/project_config.cpp

```diff
@@ -235,82 +235,87 @@ void ProjectConfig::setMDNSConfig(const std::string& hostname,
   this->config.mdns.service.assign(service);
 
   if (shouldNotify)
-    this->notify(ObserverEvent::mdnsConfigUpdated);
+    this->notifyAll(ConfigState_e::mdnsConfigUpdated);
 }
 
-void ProjectConfig::setCameraConfig(uint8_t* vflip,
-                                    uint8_t* framesize,
-                                    uint8_t* href,
-                                    uint8_t* quality,
-                                    uint8_t* brightness,
+void ProjectConfig::setCameraConfig(uint8_t vflip,
+                                    uint8_t framesize,
+                                    uint8_t href,
+                                    uint8_t quality,
+                                    uint8_t brightness,
                                     bool shouldNotify) {
```

Comment:
no, you are understanding perfectly. The issue with pass by reference is that it's only useful for data that needs to be mutated, instanced, or is quite large on the heap. Primitive types like bytes actually cause a slower performance when passed by reference due to the pointer indirection. A pointer is 4 bytes, so you end up passing 5 bytes instead of just creating a local scope of 1 byte -  plus te value to passed, so 1 more bye. These are freed when they go out of scope - so  a lot better.

Plus, it is true, that pointer indirection causes more instructions. So, while more isntructions doesn't mean worse code (less instructions doesn't mean better) - it is best practice to optimize for less instructions to try and lower the cycles to complete a computation.